### PR TITLE
Deprecates 'supports' attribute in default resource.

### PR DESCRIPTION
**Pull request by RyanFrantz**
### Description

- Per https://docs.chef.io/deprecations_property_name_collision.html, Chef 13 will bail on runs where a resource property name collides.
- Replaces 'supports' with 'handler_type' which jibes nicely with an argument name found in the helpers.
- Signed-off-by: Ryan Frantz <ryanleefrantz@gmail.com>

### Issues Resolved

Resolves Chef 13 deprecation issues described in https://docs.chef.io/deprecations_property_name_collision.html

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by RyanFrantz** [comment]
Is there anything left to complete for this PR? Note that I did not check the box for "New functionality includes testing" because this doesn't feel like new functionality; it's a simple rename. All currently defined tests in this repo passed.

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license 'Apache-2.0'
 description 'Distribute and enable Chef Exception and Report handlers'
 long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version '2.1.1'
+version '2.2.0'
```
Comment (iennae) [inline metadata.rb]: can you remove this from the commit? we do metadata updates when we do the release. Thanks!

**Comment by RyanFrantz** [comment]
Yeah, Chef 12 definitely throws a deprecation warning for this issue. While preparing for Chef 13, I updated my Test Kitchen config like so:

```
provisioner:
  ...
  deprecations_as_errors: true
```

so that chef runs fail and it's easier to address these issues (rather than trawling log files).

Do you know if there is some mechanism whereby I can tell Chef "ignore cases where `supports` is currently overridden"?

**Comment by coderanger** [comment]
@RyanFrantz Sorry, we're working on fixing but it's a false-positive deprecation because of how we did the change to `supports`.

**Comment by lamont-granquist** [comment]
if you rename it, then its a breaking change so you can't do this change as-is, period, :-1:

you could do something along these lines:

```ruby
if Gem::Requirement.new(">= 13").satisfied_by?(Gem::Version.new(Chef::VERSION))
  attribute :supports, kind_of: Hash, default: { report: true, exception: true }
end
```

mildly gross, but its self documenting that as soon as we no longer care about Chef-12 that it goes away.

**Comment by RyanFrantz** [comment]
Is there any option to tell Chef I *don't* want to bail on this deprecation warning? Something akin to Foodcritic's feature of not applying a rule (`~FC001`), say.

**Comment by tas50** [comment]
There is not at the moment.

**Comment by lamont-granquist** [comment]
To reiterate some points i was just making on slack to @iennae :  

I don't like the shape of this API as a Hash, the fact that you're setting it to true/false and the values aren't carrying anything more than boolean information indicates that an Array would probably be a better API.

I agree that the word `supports` is bad, and someone just blindly copied a bad pattern that we've removed from Chef::Resource in Chef-13 for good reason.

I think `handler_type` is probably too wordy, and that `type` would most likely be better (we know we're setting up a handler inside of the resource, and `type` would clearly refer to the handler)

So the future shape of the API should probably be:

```
  type [ :report, :exception ]
```

But there needs to be some backcompat support and `supports { report: true, exception: true }` needs to continue to work for awhile because otherwise it pretty much breaks everyone that uses this cookbook.
